### DISCUSSION TITLE
Apple Anonymous Attestation fail

### DISCUSSION THREAD

**Discussion by santhoshit8** [webauthn4j/webauthn4j#694] [General]
Hi all,

I am using Trust anchor resolution using FIDO Metadata Service.

when testing with mac touch id,  Apple Anonymous Attestation failed with error "TrustAnchors are not found".

Returned aaguid is f24a8e70-d0d3-f82c-2937-32523cc4de5a and its not found in MDS3 blob.

what should i do for Apple Anonymous Attestation?

Thanks

**Comment by ynojima** [comment]
Sorry for not documenting it.
You can find Apple WebAuthn Root CA here: https://www.apple.com/certificateauthority/private/
Please create a JKS file and import the Root CA certificate into the JKS file, and supply it to WebAuthn4J via `KeyStoreTrustAnchorRepository` class.
If you want to use it in conjection with FIDO MDS, use `AggregatingTrustAnchorRepository` to join the result from 
`KeyStoreTrustAnchorRepository` and `MetadataBLOBBasedTrustAnchorRepository`.
